### PR TITLE
[agent] Add validate_channel method

### PR DESCRIPTION
**Pull request by VladyslavTupikin**
For correct implementation Backhaul STA
Steering need to add new method in the
wireless_utils for the channel validation.

Add implementation for the method.

### CODE REVIEW

File: common/beerocks/bcl/source/son/son_wireless_utils.cpp [hunk 1]
```diff
@@ -869,3 +869,20 @@ std::list<uint8_t> wireless_utils::string_to_wsc_oper_class(const std::string &o
...
+ * @return True if channel matches to operating class
+ */
+bool wireless_utils::channel_validate(uint8_t operating_class, uint8_t channel)
+{
+    auto channel_set = operating_class_to_channel_set(operating_class);
+
+    if (channel_set.find(channel) != channel_set.end()) {
+        return true;
+    }
+    return false;
```
Comment (rmelotte) [inline common/beerocks/bcl/source/son/son_wireless_utils.cpp]:
```suggestion
    return (channel_set.find(channel) != channel_set.end())
```

**Review by rmelotte — APPROVED** [review]
Shouldn't we also check if the channel is operable (e.g. if vht capability is required)?

Or is left to the controller to only send steering requests for operable channels?
I couldn't find this information easily in the specification.

File: common/beerocks/bcl/source/son/son_wireless_utils.cpp (hunk 1 above)
Comment (VladyslavTupikin) [reply common/beerocks/bcl/source/son/son_wireless_utils.cpp]: accepted!

File: common/beerocks/bcl/include/bcl/son/son_wireless_utils.h [hunk 2]
```diff
@@ -170,6 +170,7 @@ class wireless_utils {
     static uint8_t get_operating_class_by_channel(const beerocks::message::sWifiChannel &channel);
     static std::list<sChannelPreference>
     get_channel_preferences(const beerocks::message::sWifiChannel supported_channels[]);
+    static bool channel_validate(uint8_t operating_class, uint8_t channel);
```
Comment (adam1985d) [inline common/beerocks/bcl/include/bcl/son/son_wireless_utils.h]: I think the naming should better indicate what the method is meant to do:
1. `channel_validate` can mean so many things - a `validate_channel_in_operating_class` would better explain itself
2. nitpick: it returns bool - so I prefer to have prefix `is_` to clarify we don't do an actual operation here - just perform a check. if you choose to accept this nitpick the naming would result in `is_channel_in_operating_class` - which I think is the best option in terms of self-explainable

File: common/beerocks/bcl/source/son/son_wireless_utils.cpp [hunk 3]
```diff
@@ -869,3 +869,17 @@ std::list<uint8_t> wireless_utils::string_to_wsc_oper_class(const std::string &o
         return {};
     }
 }
+
+/**
+ * @brief convert operating class to bandwidth based on Table 4-E in the ieee 802.11 specification
```
Comment (adam1985d) [inline common/beerocks/bcl/source/son/son_wireless_utils.cpp]: probably a copy-paste typo? the method brief description is incorrect

**Review by adam1985d — CHANGES_REQUESTED** [review]

File: common/beerocks/bcl/include/bcl/son/son_wireless_utils.h (hunk 2 above)
Comment (VladyslavTupikin) [reply common/beerocks/bcl/include/bcl/son/son_wireless_utils.h]: accepted!

File: common/beerocks/bcl/source/son/son_wireless_utils.cpp (hunk 3 above)
Comment (VladyslavTupikin) [reply common/beerocks/bcl/source/son/son_wireless_utils.cpp:874]: This is not a related issue.

Comment (VladyslavTupikin) [reply common/beerocks/bcl/source/son/son_wireless_utils.cpp]: Accepted! Moved Doxygen to the header file, according to the poll results in democracy conversation.

**Review by adam1985d — APPROVED** [review]

**Pull request merged**